helix-core/src/syntax.rs

```diff
@@ -769,30 +772,38 @@ impl Syntax {
         // Convert the changeset into tree sitter edits.
         let edits = generate_edits(old_source, changeset);
 
+        // This table allows inverse indexing of `layers`.
+        // That is by hashing a `Layer` you can find
+        // the `LayerId` of an existing equivalent `Layer` in `layers`.
+        //
+        // It is used to determine if a new layer exists for an injection
+        // or if an existing layer needs to be updated.
+        let mut layers_table = RawTable::with_capacity(self.layers.len());
+        let layers_hasher = RandomState::new();
```

Comment:
Why do we need to pull in `RawTable`? Why not just `HashMap::with_capacity()`?

Reply:
The point of the `RawTable` is not the `capacity`/`insert_no_grow`. That mainly just a nice bonus.
A `HashMap<K,V>` is essentially a `RawTable<(K,V)>` internally. However we want a `LanguageLayer -> LayerId` map here so using a normal `HashMap<LanguageLayer, LayerId>` would require that we clone all `LanguageLayer`s (we can not store references into `Syntax::layers` because it is mutated later).
With a `RawTable` we can just store the `LayerId` as `RawTable<LayerId>` .
When we need access to the original keys to resolve hash collisions during lookup later we can just use the `LayerId` to index back into `self.layers`.

`IndexMap` does essentially the same thing  internally (it just doesn't fit well here because we want to keep using a `HopSlotMap` whereas `indexmap` forces us to use essentially a `Vec`).

I also use the exact same strategy with `RawTable` for the interning in `imara-diff` (to avoid pulling in `indexmap` which offers a lot more functionality). So we will depend on `hashbrown` anyway (in fact we already do so as a dependy of `unicode-linebreak`)